File: src/main/java/io/github/richardson/ObjectManager.java
package io.github.richardson;

import io.github.richardson.menu.Settings;
import processing.core.PApplet;

public class ObjectManager {
    public final Settings settings;
    public final Key key;
    public final Cursor cursor;
    public final Font font;
    private final PApplet p;

    public ObjectManager(PApplet p) {
        this.p = p;
        this.settings = new Settings();
        this.key = new Key(p, settings);
        this.cursor = new Cursor(p, 3000, true);
        this.font = new Font(p, "fonts/JetBrainsMono-VariableFont_wght.ttf", 32);
    }
}

File: src/main/kotlin/io/github/richardson/ObjectManager.kt
package io.github.richardson

import io.github.richardson.menu.Settings
import processing.core.PApplet

class ObjectManager(p: PApplet) {
    @JvmField
    val settings: Settings
    @JvmField
    val key: Key
    @JvmField
    val cursor: Cursor
    @JvmField
    val font: Font
    private val p: PApplet?

    init {
        this.p = p
        this.settings = Settings()
        this.key = Key(p, settings)
        this.cursor = Cursor(p, 3000, true)
        this.font = Font(p, "fonts/JetBrainsMono-VariableFont_wght.ttf", 32f)
    }
}